tests/Sharpliner.Tests/AzureDevOps/Pipelines/PlaygroundPipeline.cs

```diff
@@ -0,0 +1,176 @@
+            Variable("Configuration", "Release"),     // Or we have this more YAML-like definition
+            Group("PR keyvault variables"),
+            If.IsPullRequest
+                .Variable("TargetBranch", variables.System.PullRequest.SourceBranch)
+                .Variable("IsPr", true),
+            If.And(IsBranch("production"), NotEqual("Configuration", "Debug"))
+                .Variable("PublishProfileFile", "Prod")
+                .If.IsNotPullRequest
+                    .Variable("AzureSubscription", "Int")
+                    .Group("azure-int")
+                .EndIf
+                .If.IsPullRequest
+                    .Variable("AzureSubscription", "Prod")
+                    .Group("azure-prod"),
+        },
+
+        Stages =
+        [
+            Stage("First") with
+            {
+                Jobs =
+                [
+                    new JavaTemplate(new()
+                    {
+                        JavaVersion = "11",
+                        MavenPomFile = "pom.xml",
+                    })
+                ]
+            }
+        ]
+    };
+}
+
+[SharplinerTemplateParameters]
+partial class JavaTemplate : JobTemplateDefinition<JavaTemplateParameters>
+{
+    public JavaTemplate() : base() {}
+    public JavaTemplate(JavaTemplateParameters parameters) : base(parameters)
+    {
+    }
+
+    public override string TargetFile => "java-template.yml";
+    public override ConditionedList<JobBase> Definition =>
+    [
+        Job("Build") with
+            {
+                Steps =
+                [
+                    Task("UseJavaVersion@0") with
+                    {
+                        Inputs = new()
+                        {
+                            ["versionSpec"] = parameters.JavaVersion,
+                            ["jdkArchitecture"] = "x64",
+                        }
+                    },
+                    Task("Maven@3") with
+                    {
+                        Inputs = new()
+                        {
+                            ["mavenPomFile"] = parameters.MavenPomFile,
+                            ["mavenOptions"] = parameters.MavenOptions,
+                            ["javaHomeOption"] = "JDKVersion",
+                            ["jdkVersionOption"] = parameters.JavaVersion,
+                            ["jdkArchitectureOption"] = "x64",
+                            ["publishJUnitResults"] = "false",
+                            ["testResultsFiles"] = "**/surefire-reports/TEST-*.xml",
+                            ["goals"] = "package",
+                            ["spring.profiles.active"] = parameters.Spring.SpringProfile,
+                            ["spring.options"] = parameters.Spring.SpringOptions,
+                        }
+                    }
+                ]
+            }
+    ];
+}
+
+class JavaTemplateParameters : TemplateParametersProviderBase<JavaTemplateParameters>
```

Comment:
I'm wondering if it would be cleaner to generate the `List<Parameter> Parameters` & `TemplateParameters` and not have to extend this base class - then users will only need to annotate the template-definition class with `[SharplinerTemplateParameters]`